mountpoint-s3-client/src/mock_client.rs

```diff
@@ -783,6 +810,15 @@ impl Stream for MockGetObjectResponse {
     type Item = ObjectClientResult<GetBodyPart, GetObjectError, MockClientError>;
 
     fn poll_next(mut self: Pin<&mut Self>, _cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        if self.fail_on_non_aligned_read_window && self.read_window_increment_failed.load(Ordering::SeqCst) {
```

Comment:
Wouldn't it be simpler to peek inside the `MockBackpressureHandle`, if present, rather than duplicating the failed flag?